### ISSUE TITLE
Table: Join multiple queries

### ISSUE THREAD

**Issue by 7840vz** [perses/perses#2341]
### Proposal

Table in grafana:
<img width="1031" alt="image" src="https://github.com/user-attachments/assets/a75c6279-d885-4880-8da3-f5f59f031933">

Table in perses:
<img width="1209" alt="image" src="https://github.com/user-attachments/assets/87937b96-3df8-4ecd-9443-266447a9e6c4">

There are two queries (for `{{mountpoint}} total` and `{{mountpoint}} free`) in Grafana with identical labels. After 'group by' transformation and 'merge' is applied, those two metrics can be seen on the same row. However, in Perses there are 4 rows instead, two (`{{mountpoint}} total` and `{{mountpoint}} free` separately) for each mountpoint. 

Would be great if it is possible to JOIN/GROUP BY specific label(s), similar to https://grafana.com/docs/grafana/latest/panels-visualizations/query-transform-data/transform-data/#group-by .

Grafana panel code:
```
{
  "datasource": {
    "type": "prometheus",
    "uid": "${datasource}"
  },
  "description": "Disk utilisation in percent, by mountpoint. Some duplication can occur if the same filesystem is mounted in multiple locations.",
  "fieldConfig": {
    "defaults": {
      "custom": {
        "align": "auto",
        "cellOptions": {
          "type": "auto"
        },
        "inspect": false
      },
      "mappings": [],
      "unit": "bytes"
    },
    "overrides": [
      {
        "matcher": {
          "id": "byName",
          "options": "Mounted on"
        },
        "properties": [
          {
            "id": "custom.width",
            "value": "260"
          }
        ]
      },
      {
        "matcher": {
          "id": "byName",
          "options": "Size"
        },
        "properties": [
          {
            "id": "custom.width",
            "value": "80"
          }
        ]
      },
      {
        "matcher": {
          "id": "byName",
          "options": "Used"
        },
        "properties": [
          {
            "id": "custom.width",
            "value": "80"
          }
        ]
      },
      {
        "matcher": {
          "id": "byName",
          "options": "Available"
        },
        "properties": [
          {
            "id": "custom.width",
            "value": "80"
          }
        ]
      },
      {
        "matcher": {
          "id": "byName",
          "options": "Used, %"
        },
        "properties": [
          {
            "id": "custom.cellOptions",
            "value": {
              "type": "gauge",
              "mode": "basic"
            }
          },
          {
            "id": "max",
            "value": 1
          },
          {
            "id": "min",
            "value": 0
          },
          {
            "id": "unit",
            "value": "percentunit"
          }
        ]
      }
    ]
  },
  "gridPos": {
    "h": 8,
    "w": 12,
    "x": 12,
    "y": 20
  },
  "id": 19,
  "pluginVersion": "11.0.0",
  "targets": [
    {
      "datasource": {
        "type": "prometheus",
        "uid": "${datasource}"
      },
      "expr": "node_filesystem_size_bytes{instance=~\"$instance\"}",
      "format": "table",
      "instant": true,
      "refId": "TOTAL"
    },
    {
      "datasource": {
        "type": "prometheus",
        "uid": "${datasource}"
      },
      "expr": "node_filesystem_avail_bytes{instance=~\"$instance\"}",
      "format": "table",
      "instant": true,
      "legendFormat": "{{ mountpoint }} free",
      "refId": "FREE"
    }
  ],
  "title": "Disk space usage",
  "transformations": [
    {
      "id": "groupBy",
      "options": {
        "fields": {
          "Value #FREE": {
            "aggregations": [
              "lastNotNull"
            ],
            "operation": "aggregate"
          },
          "Value #TOTAL": {
            "aggregations": [
              "lastNotNull"
            ],
            "operation": "aggregate"
          },
          "mountpoint": {
            "aggregations": [],
            "operation": "groupby"
          }
        }
      }
    },
    {
      "id": "merge",
      "options": {}
    },
    {
      "id": "calculateField",
      "options": {
        "alias": "Used",
        "binary": {
          "left": "Value #TOTAL (lastNotNull)",
          "operator": "-",
          "reducer": "sum",
          "right": "Value #FREE (lastNotNull)"
        },
        "mode": "binary",
        "reduce": {
          "reducer": "sum"
        }
      }
    },
    {
      "id": "calculateField",
      "options": {
        "alias": "Used, %",
        "binary": {
          "left": "Used",
          "operator": "/",
          "reducer": "sum",
          "right": "Value #TOTAL (lastNotNull)"
        },
        "mode": "binary",
        "reduce": {
          "reducer": "sum"
        }
      }
    },
    {
      "id": "organize",
      "options": {
        "excludeByName": {},
        "indexByName": {
          "Used": 3,
          "Used, %": 4,
          "Value #FREE (lastNotNull)": 2,
          "Value #TOTAL (lastNotNull)": 1,
          "mountpoint": 0
        },
        "renameByName": {
          "Value #FREE (lastNotNull)": "Available",
          "Value #TOTAL (lastNotNull)": "Size",
          "mountpoint": "Mounted on"
        }
      }
    },
    {
      "id": "sortBy",
      "options": {
        "fields": {},
        "sort": [
          {
            "desc": false,
            "field": "Mounted on"
          }
        ]
      }
    }
  ],
  "type": "table",
  "options": {
    "showHeader": true,
    "cellHeight": "sm",
    "footer": {
      "show": false,
      "reducer": [
        "sum"
      ],
      "countRows": false,
      "fields": ""
    }
  }
}
```

**Comment by AntoineThebaud** [comment]
This is actually a feature already planned on our side for the near future. Assigning this one to @Gladorme.

**Comment by Gladorme** [comment]
Yup, I confirm I am currently working on it 😄

**Comment by Gladorme** [comment]
Implemented with #2359, I think we can close this issue 😄
